hypervisor/src/mshv/mod.rs

```diff
@@ -884,6 +884,22 @@ impl vm::Vm for MshvVm {
             "get_dirty_log not implemented"
         )))
     }
+    ///
+    /// Enable dirty page tracking
+    ///
+    fn enable_dirty_page_tracking(&self) -> vm::Result<()> {
+        self.fd
+            .enable_dirty_page_tracking()
```

Comment:
Will this enable dirty-page-tracking on all guest memory regions? 

With KVM, we are now enabling dirty-page-tracking only on memory regions for guest RAM and excluding regions backed by device/file, e.g. virtio-fs, pmem, SGX. Will mshv be able to do something similar?

/cc @rbradford @sboeuf

Reply:
Tracking should work for any page mapped to the guest's GPA space. That means whatever memory we map through set_user_memory should be tracked.

Reply:
How does the map of these types of memory work in CLH?

Reply:
There is no need to track all pages mapped to the guest's GPA space, and we only need to kvm/mshv to track dirty pages for guest RAM. This is how CLH is working with kvm by using mappings from `MemoryManager::guest_ram_mappings`.

Can you make the `mshv` crate provide a API so that it can enable dirty page logging on a slot/region basis (similar to what `kvm_ioctls::ioctls::vm::VmFd::get_dirty_log()` provides)?

Reply:
Right now Microsoft Hypervisor  does not have such API.